upt_macports/upt_macports.py

```diff
@@ -13,7 +15,41 @@ def __init__(self):
     def create_package(self, upt_pkg, output):
         self.upt_pkg = upt_pkg
         self.logger.info(f'Hello, creating the package')
-        print(self._render_makefile_template())
+        if output is None:
+            print(self._render_makefile_template())
+        else:
+            self._create_output_directories(upt_pkg, output)
+            self.logger.info('Directory structure created')
+            self._create_portfile()
+
+    def _create_output_directories(self, upt_pkg, output_dir):
+        """Creates the directory layout required"""
+        self.logger.info(f'Creating the directory structure in {output_dir}')
+        upt2macports = {
+            'cpan': 'perl',
+            'pypi': 'python',
+            'rubygems': 'ruby',
+        }
+        try:
+            port_category = upt2macports.get(self.upt_pkg.frontend)
+        except KeyError:
+            sys.exit(
+                f'Unknown port category for frontend {self.upt_pkg.frontend}')
+        folder_name = self._normalized_macports_folder(upt_pkg.name)
+        self.output_dir = os.path.join(
+            output_dir, 'ports', port_category, folder_name)
```

Comment:
I am not sure about this... why is there a hard-coded 'ports' in the ```output_dir```? I assume that most developers will have a [local copy of the ports-tree](https://guide.macports.org/chunked/development.local-repositories.html); for example I have mine in ```~/Software/macports-ports```. In that case I should be able to use that as my "ouput" directory and that doesn't work right now.

Reply:
I believe one of the tools used the `dports` prefix, but I don't like that either.